data/basal/sample_builder_test.go

```diff
@@ -0,0 +1,105 @@
+package basal
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/mdblp/tide-whisperer-v2/v2/schema"
+	schemaV1 "github.com/tidepool-org/tide-whisperer/schema"
+)
+
+
+func Test_makeBasalSample(t *testing.T) {
+	timeStart := time.Date(2023, time.March ,14, 0, 0, 0, 0, time.UTC)
+	timeEnd := time.Date(2023, time.March, 14, 1, 0, 0, 0, time.UTC)
+	sample := makeBasalSample(timeStart, timeEnd, "test", 
+		schema.BasalSample{
+			Sample: schema.Sample{
+				Timestamp: time.Now(),
+				Timezone: "Europe/Paris",
+				TimezoneOffset: 0,
+			},
+			Guid: "Test",
+			DeliveryType: "not a test",
+			Rate: 1.5,
+			Duration: 60000,
+		})	
+	expected := schema.BasalSample{
+		Sample: schema.Sample{
+			Timestamp: timeStart,
+			Timezone: "Europe/Paris",
+			TimezoneOffset: 60,
+		},
+		Guid: "Test",
+		DeliveryType: "test",
+		Rate: 1.5,
+		Duration: 60 * 60 * 1000,
+	}
+
+	test := reflect.DeepEqual(sample, expected)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:**  why not using table testing ?‏